cumulusci/robotframework/CumulusCI.py

```diff
@@ -114,6 +116,46 @@ def login_url(self, org=None):
             org = self.keychain.get_org(org)
         return org.start_url
 
+    def _refresh_community_info_cache(self):
+        response = self.sf.restful("connect/communities")
+        self._community_info_cache = {
+            community["name"]: community for community in response["communities"]
+        }
+
+    def get_community_info(self, community_name, key=None, force_refresh=False):
+        """This keyword uses the Salesforce API to get information about a community.
+
+        This keyword requires the exact community name as its first argumment.
+        - If no key is given, all of the information returned by the API will be
+          returned by this keyword in the form of a dictionary
+        - If a key is given, only the value for that key will be returned.
+
+        Some of the supported keys include name, siteUrl, and
+        loginUrl. For a comprehensive list see the
+        [https://developer.salesforce.com/docs/atlas.en-us.chatterapi.meta/chatterapi/connect_responses_community.htm|API documentation],
+        or call this keyword without the key argument and examine the
+        results.
+
+        An API call will be made the first time this keyword is used, and
+        the return values will be cached. Subsequent calls will not call
+        the API unless the requested community name is not in the cached
+        results, or unless the force_refresh parameter is set to True.
+        """
+        if force_refresh or community_name not in self._community_info_cache:
+            self._refresh_community_info_cache()
+
+        if community_name not in self._community_info_cache:
+            raise Exception(
+                "Unable to find community information for '{}'".format(community_name)
+            )
+
+        if key is None:
+            return self._community_info_cache[community_name]
+        else:
+            if key not in self._community_info_cache[community_name]:
+                raise Exception("Invalid key '{}'".format(key))
+            return self._community_info_cache[community_name][key]
```

Comment:
It might make sense to put most of this logic in OrgConfig so that another task that needs community info could make use of it. There's nothing robot-specific about it.

Reply:
I moved this code into OrgConfig. Thanks for the suggestion!